python/ray/dag/compiled_dag_node.py

```diff
@@ -1694,7 +1784,18 @@ def _get_or_compile(
                 executable_tasks.append(executable_task)
             # Sort executable tasks based on their bind index, i.e., submission order
             # so that they will be executed in that order.
-            executable_tasks.sort(key=lambda task: task.bind_index)
+            executable_tasks.sort(
+                # If the bind index is the same, there are P2P send/recv tasks.
+                # The order is determined as follows:
+                # 1. P2P recv tasks.
```

Comment:
I think this is quite brittle and difficult to understand. It would be better if we can think of a way to make all of the bind indices for a given actor unique.

Either we need to compute the new bind indices when creating the CompiledDAG or we need a way to create the NCCL nodes while the user is creating the initial DAG.